lib/MediaConnection/MediaConnection.js

```diff
@@ -0,0 +1,155 @@
+const EventEmitter = require('events').EventEmitter;
+
+module.exports = class MediaConnection extends EventEmitter
+{
+  constructor(pcConfig, iceConnectionStateCallback, rtcPeerConnection = new RTCPeerConnection(pcConfig))
```

Comment:
No need for `iceConnectionStateCallback`. This class already exposes the `addEventListener()` method, plus it accepts an optional `RTCPeerConnection` instance in construction. So if you are interested in acting upon ice connection state change you can do it directly on the optional `RTCPeerConnection` passed as argument or use the public method of this class `addEventListener()`.

```suggestion
  constructor(pcConfig, rtcPeerConnection = new RTCPeerConnection(pcConfig))
```